Move attestation model into the site

joshuagl opened the pull request:
This moves the attestation model, referenced from the Provenance and VSA pages, into the main site so we are not bouncing readers who follow the link to the GitHub repository.


Fixes #323

joshuagl commented:
Link to the new page on the preview site: https://deploy-preview-438--slsa.netlify.app/spec/attestation-model

MarkLodato commented on controls/attestations.md:
```diff
@@ -1 +0,0 @@
-Content moved to [../docs/attestation-model.md](../docs/attestation-model.md).
```
Let's leave this here until we update all old references, notably in the in-toto/attestation and in-toto/ITE repositories.

joshuagl replied:
Yes, that makes sense. I ended up `rm`ing it because the linter complained, but I can add it back and placate the linter.

MarkLodato commented on docs/spec/attestation-model.md:
```diff
@@ -1,97 +1,88 @@
 ---
```
What is the motivation from moving it into the specification? Isn't it independent from the level specification? Your original location (top-level docs/) better aligned with my expectations, but I may be missing something.

joshuagl replied:
Simply that it felt odd linking to it from the specification menu when it wasn't tracked as part of the spec, but I was not convinced. As you aren't convinced either, I will move it back.

MarkLodato commented on docs/_includes/nav.html:
```diff
@@ -37,7 +37,7 @@
                   <hr>
                   {%- else %}
-                  <a class="text-white md:text-black" href="{{ subitem.url | relative_url }}">
+                  <a class="text-white md:text-black {% if subitem.nudge %}pl-5{% endif %} " href="{{ subitem.url | relative_url }}">
```
As an alternative, you could do something like this, adding an unlinked "Attestations:" heading and then link to the model as a sub-page of that:

![image](https://user-images.githubusercontent.com/58860/180481271-bafbf8d7-d621-4913-9a94-dca90629a1db.png)

joshuagl replied:
That's less hacky, let me do that.

MarkLodato replied:
In my screenshot I also added a `margin: 13px 0` to the `<li>` containing the horizontal rule so that it's not so smushed.

MarkLodato reviewed (changes requested):
This is great, thanks! Not only is the new location is helpful, but the wording and formatting changes are also a big readability improvement.

I do have a few small suggestions, but overall LGTM.

joshuagl commented:
Thanks for the quick review Mark. I've addressed your comments as follow-on commits. I can clean up the git history before merging, or squash and merge – I don't have a preference but thought follow-on commits would be easier to review.

MarkLodato commented on docs/attestation-model.md:
```diff
@@ -1,96 +1,88 @@
 title: Software attestations
-layout: standard
-hero_text: A software attestation is a signed statement (metadata) about a software artifact or collection of software artifacts.
+version: 0.1
```
Is there any way to easily get rid of the version dropdown? It doesn't make sense for this page. (See [preview](https://deploy-preview-438--slsa.netlify.app/attestation-model)))

joshuagl replied:
I will play around with this, perhaps changing the layout back to `standard`.

joshuagl replied:
If I switch back to standard layout I get the large green header section for the hero text. Formatting text within that section is... awkward. So I've ended up changing the specification layout to _not_ show the version selector for the "Software attestations" page, as has already been done for the FAQ and use-cases pages.

MarkLodato commented on docs/attestation-model.md:
```diff
@@ -1,96 +1,88 @@
-![Attestation model diagram](../images/attestation_layers.svg)
-
-Example in English:
+![Attestation model diagram](/images/attestation_layers.svg)
```
nit: can you use relative paths to the images so that it renders correctly on the GitHub preview?

Optionally you can do the same for Markdown links (but not HTML links) since the Jekyll plugin convers links to .md files to the proper URL. But the value is much lower. Example: docs/roadmap.md, line 11

joshuagl replied:
I will fix the images to use relative paths and try to improve the non-image links too.

joshuagl commented on controls/attestations.md:
```diff
@@ -1 +1,5 @@
+
+This content is now available at
+[../docs/attestation-model.md](../docs/attestation-model.md) and on
+[the website](https://slsa.dev/attestation_model).
```
One day I'll stop pushing last minute changes on a Friday, but today is not that day. Apologies and thanks for the diligent review. I'll fix this, and the rest of the PR, on Monday.

MarkLodato commented on controls/attestations.md:
```diff
@@ -1,5 +1,5 @@
 This content is now available at
-[../docs/attestation-model.md](../docs/attestation-model.md) and on
-[the website](https://slsa.dev/attestation_model).
+[../docs/attestation_model.md](../docs/attestation-model.md) and on
```
Oops! Shouldn't it always be a hyphen?

joshuagl replied:
It should 🤦 . Fixed in 9b52b49

(this took an embarrassing number of attempts to get right 😆 )

MarkLodato reviewed (approved):
LGTM. Thanks!

joshuagl commented:
Any opinions on rebase vs. squash vs. leave as-is with messy PRs like this?

MarkLodato commented:
> Any opinions on rebase vs. squash vs. leave as-is with messy PRs like this?

I'd say rebase if you're up for it, or merge as-is if you don't care. Other PRs are much messier and I've learned to live with an ugly git history (GitHub's model seems to encourage that.)

I don't particularly like the squash option since (IIRC) GitHub provides a barebones commit message and it's too much work for me to recreate a nice commit message. But I'm not opposed to it.

joshuagl commented:
I've rebased to clean up commit history, the original (unclean) set of changes is still visible at https://github.com/joshuagl/slsa/commits/joshuagl/attestation-model-unclean

joshuagl commented:
I'll leave this for another day for folks to comment, then merge if there aren't any objections or major changes requested.
